omc3/scripts/kmod_summary.py

```diff
@@ -76,144 +99,235 @@ def output_kmod_summary_tables(
-    LOG.info("Starting kmod summary importing.")
-    cols_x = [f"{BETASTAR}X", f"{ERR}{BETASTAR}X", f"{BETAWAIST}X", f"{ERR}{BETAWAIST}X", f"{WAIST}X", f"{ERR}{WAIST}X"]
-    cols_y = [f"{BETASTAR}Y", f"{ERR}{BETASTAR}Y", f"{BETAWAIST}Y", f"{ERR}{BETAWAIST}Y", f"{WAIST}Y", f"{ERR}{WAIST}Y"]
-    beams = [f"{BEAM_DIR}{1}", f"{BEAM_DIR}{2}"]
 
-    grouped = {beam: [] for beam in beams}
+    LOG.info(f"Starting kmod summary importing for {BEAM_DIR}{beam}.")
+    grouped = _collect_kmod_results(meas_paths=meas_paths, beam=beam)
+
+    if averaged_meas is not None:
+        grouped_averaged = _collect_averaged_kmod_results(averaged_meas=averaged_meas, beam=beam)
+
+    if lumi_imbalance:
+        kmod_summary_lumiimb = _get_lumi_imbalance(output_dir=output_dir)
+    else:
+        LOG.info("Luminosity imbalance calculation skipped.")
+
+    LOG.debug(f"Processing result for: {beam}")
+    if grouped:
+        kmod_summary = tfs.concat(grouped, ignore_index=True)
+        if averaged_meas:
+            kmod_summary_averaged = tfs.concat(grouped_averaged, ignore_index=True)
+        table_logbook = _prepare_logbook_table(beam=beam, kmod_summary = kmod_summary, kmod_summary_averaged = kmod_summary_averaged, kmod_summary_lumiimb=kmod_summary_lumiimb)
+
+    if output_dir is not None:
+        _save_outputs(beam=beam, save_output_dir=output_dir, table_logbook=table_logbook, summary_df=kmod_summary)
+    else:
+        LOG.info("Output_dir not provided: skipping saving files for all beams.")
+
+    if logbook:
+        _logbook_entry(beam=beam, logbook=logbook, logbook_file=table_logbook)
+    if not logbook:
+        LOG.info("Logbook name not provided: logbook entry not created.")
+
+    return kmod_summary, table_logbook
+
+def _collect_kmod_results(
+    meas_paths: Sequence[Path | str],
+    beam: int,
+    ) -> list[tfs.TfsDataFrame]:
+
     """
     Gathers the various Kmod results.tfs dataframes, taking only cols_x and cols_y values.
-    Then it stores them in grouped, divided by beam.
+
+    Args:
+        meas_paths (Sequence[Path | str]): List of measurement directories containing beam subfolders.
+        beam (int): Beam number to process.
+
+    Returns:
+        list[tfs.TfsDataFrame]: List containing grouped kmod results.
     """
+
+    LOG.info("Grouping kmod results.")
+
+    grouped = []
     for path in meas_paths:
         path = Path(path)
-        LOG.info(f"Reading results: {path}.")
-        for beam in beams:
-            file_path = path / beam / "results.tfs"
-            if not file_path.exists():
-                LOG.warning(f"Missing results.tfs: {file_path}")
+        LOG.info(f"Reading results: {path.absolute()}.")
+        file_path = path / f"{BEAM_DIR}{beam}" / f"{RESULTS_FILE_NAME}{EXT}"
+        if not file_path.exists():
+            LOG.warning(f"Missing {RESULTS_FILE_NAME}{EXT}: {file_path}")
+        else:
+            meas_name = file_path.parent.parent.name
+            data_file = tfs.read(file_path)
```

Comment:
It's not a file, it's a dataframe. Please rename. You can name it `df` or `result_df` if you want.